lib/v2/taptap/topic.js

```diff
@@ -33,22 +33,29 @@ module.exports = async (ctx) => {
     const topics_list_response = await got(url);
     const topics_list = topics_list_response.data;
 
-    const app_img = topics_list.data.list[0].moment.app.icon.url;
-    const app_name = topics_list.data.list[0].moment.app.title;
+    let app_img = null;
+    let app_name = null;
 
     const out = topics_list.data.list.map(async (list) => {
+        if (app_img === null && app_name === null) {
```

Comment:
This condition is always true. You can skip the outer `if`

Reply:
>This condition is always true.

No.

I've changed app_img/app_name from a constant into a variable, and it's outside the `.map(...)` loop.

once those two variables has been set properly, they shouldn't be set again. so there is an `if` here.